starknet_py/net/client_test.py

```diff
@@ -19,15 +20,27 @@
 @pytest.mark.asyncio
 async def test_wait_for_tx_throws_transaction_rejected():
     client = Client("testnet")
+    code = "TRANSACTION REJECTED"
+    message = "Transaction was rejected"
 
     result = Future()
-    result.set_result(MagicMock(status=TransactionStatus.REJECTED))
+    result.set_result(
```

Comment:
I though it'd be better to test the logic of this function using mocks rather than rely on the responses from starknet. If you want actual tests for these, we should probably move all these tests to e2e tests and actually run them on devnet.

Reply:
let's create a task to make similar tests run on testnet when needed. we can leave this one as-is